src/Microsoft.AspNetCore.Mvc.Core/ApplicationModels/SelectorModel.cs

```diff
@@ -12,6 +12,7 @@ public class SelectorModel
         public SelectorModel()
         {
             ActionConstraints = new List<IActionConstraintMetadata>();
+            EndpointMetadata = new List<object>();
```

Comment:
Yeah? I just followed the pattern with constraints and got metadata from the selector model. Do you have a better idea?

Reply:
In the past when we've added things to application model we've added them at all levels of heirarchy. It's somewhat complicated to reason about and a little annoying to implement. 

This allows you to (for example) apply a filter at the controller level or global level. I wish I would have design application model in 1.0 as a flat list of `action` objects instead of a hierarchy. It's easy to write a linq query and a `foreach`. This is the regret that I'm referring too. 

I don't feel like it's needed in this case - what you did is fine.